appinventor/components/src/com/google/appinventor/components/runtime/Camera.java

```diff
@@ -176,6 +191,56 @@ public void onGranted() {
     }
   }
 
+  @SuppressLint("NewApi")
+  @SuppressWarnings("deprecation")
+  @SimpleFunction(description = "Toggle the flash of your device to on or off.")
+  public void ToggleLight() {
+    if (!havePermission) {
+      form.askPermission(CAMERA,
```

Comment:
I failed to find an alternate implementation that works across different devices and API levels. Modern solutions (CameraX) require API level 21+.

Reply:
Right. Normally we'd introduce something conditional on SDK into the corresponding Util class (in this case LollipopUtil). However, because the Camera reference is stateful that approach won't work here. I think this is fine as is my concern was mainly with the permission part of it given that if FLASHLIGHT is a thing then maybe it would be more contextually relevant to use, but obviously if it's not then CAMERA makes plenty of sense.